stumpy/floss.py

```diff
@@ -626,6 +699,12 @@ def P_(self):
     def I_(self):
         """
         Get the updated (right) matrix profile indices
+
+        Note1: In contrast to the output of `stump`, any invalid index `-1`
+        in this (right) matrix profile indices is replaced with the start
+        index of the corresponding subsequence.
+        Note2: The indices stored in `self.I_` reflect the starting index of
+        subsequneces with respect to the original time series.
```

Comment:
I have also added two notes here that can help user better understand the information stored in `self._I`

Reply:
You don't need `Note1` and `Note2`:

> The right matrix profile indices reflect the starting index of subsequences with respect to the original time series and all invalid matrix profile indices are replaced with the start index of the corresponding subsequence (rather than with `-1`).

I can't remember if we should/ned to keep bad indices as the "self index". I think we can/should convert the index back to `-1` as a post-processing step (and then we can omit the second half of the note above). However, we should make a copy first and then hand it back to the user

Reply:
@seanlaw 
> I can't remember if we should/ned to keep bad indices as the "self index".

Me neither :) I think I wanted to do that but after realizing the concept behind it, I did not think about it again.

> I think we can/should convert the index back to `-1` as a post-processing step

As you suggested in #808, we can do:
```
I = self._mp[:, 3].astype(np.int64).copy()
I[self._mp[:, 0] == np.inf] = -1
```

I might have some thoughts regarding the `self._I` that I can share with you in the future when I finish my investigation. 

For now, I think we should be good. I will revise the code.

Reply:
I think this should say "with respect to the full time series (i.e., including all egressed data points)"